api/controllers/datasets.py

```diff
@@ -134,6 +143,16 @@ def create(credentials, tid, name):
             logger.exception(ex)
             bottle.abort(400, "Invalid dataset file")
 
+    # Load tag hierarchy (if it exists)
```

Comment:
maybe we can add some verification logic, to make sure the tag hierarchy is in the right form here?